src/volttron/client/commands/control.py

```diff
@@ -706,34 +709,50 @@ def disable_agent(opts):
 
 
 def start_agent(opts):
-    call = opts.connection.call
-    agents = _list_agents(opts)
-    for pattern, match in filter_agents(agents, opts.pattern, opts):
-        if not match:
-            _stderr.write("{}: error: agent not found: {}\n".format(opts.command, pattern))
-        for agent in match:
-            pid, status = call("agent_status", agent.uuid)
-            if pid is None or status is not None:
-                _stdout.write("Starting {} {}\n".format(agent.uuid, agent.name))
-                call("start_agent", agent.uuid)
+    act_on_agent("start_agent", opts)
 
 
 def stop_agent(opts):
+    act_on_agent("stop_agent", opts)
+
+
+def restart_agent(opts):
+    stop_agent(opts)
+    start_agent(opts)
+
+
+def act_on_agent(action, opts):
```

Comment:
Please document the new "act_on_agent" function now including what is expected in opts and how this method works.